src/aiida/orm/nodes/data/remote/base.py

```diff
@@ -185,3 +198,105 @@ def _validate(self):
 
     def get_authinfo(self):
         return AuthInfo.get_collection(self.backend).get(dbcomputer=self.computer, aiidauser=self.user)
+
+    def get_size_on_disk(self, relpath: Path | None = None) -> str:
+        """
+        Connects to the remote folder and returns the total size of all files in the directory recursively in a
+        human-readable format.
+
+        :param relpath: File or directory path for which the total size should be returned, relative to
+        ``self.get_remote_path``.
+        :return: Total size of file or directory in human-readable format.
+
+        :raises: FileNotFoundError, if file or directory does not exist.
+        """
+
+        from aiida.common.utils import format_directory_size
+
+        if relpath is None:
+            relpath = Path('.')
+
+        authinfo = self.get_authinfo()
+        full_path = Path(self.get_remote_path()) / relpath
+        computer_label = self.computer.label if self.computer is not None else ''
+
+        with authinfo.get_transport() as transport:
+            if not transport.path_exists(str(full_path)):
+                exc_message = (
+                    f'The required remote folder {full_path} on Computer <{computer_label}>'
+                    'does not exist, is not a directory or has been deleted.'
+                )
+                raise FileNotFoundError(exc_message)
+
+            try:
+                total_size: int = self._get_size_on_disk_du(full_path, transport)
+
+            except RuntimeError:
+                lstat_warn = (
+                    'Problem executing `du` command. Will return total file size based on `lstat`. '
+                    'Take the result with a grain of salt, as `lstat` does not consider the file system block size, '
+                    'but instead returns the true size of the files in bytes, which differs from the actual space'
+                    'requirements on disk.'
+                )
+                _logger.warning(lstat_warn)
+
+                total_size: int = self._get_size_on_disk_lstat(full_path, transport)
+
+            except OSError:
+                _logger.critical('Could not evaluate directory size using either `du` or `lstat`.')
+
+        return format_directory_size(size_in_bytes=total_size)
+
+    def _get_size_on_disk_du(self, full_path: Path, transport: 'Transport') -> int:
+        """Connects to the remote folder and returns the total size of all files in the directory recursively in bytes
+        using `du`.
+
+        :param full_path: Full path of which the size should be evaluated
+        :type full_path: Path
+        :param transport: Open transport instance
+        :type transport: Transport
+        :raises RuntimeError: When `du` command cannot be successfully executed
+        :return: Total size of directory recursively in bytes.
```

Comment:
Sorry, I found the term recursively confusing.. (thought the function does  `yield` )
```suggestion
        :return: Total size of directory in bytes (including all it's contents)
```